Replace MarkdownSharp with Markdig

StephenMcConnel opened the pull request:
Also remove apparently unused reference to MarkdownDeep.

This is the first step of the new localization approach to markdown files.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1793)
<!-- Reviewable:end -->

JohnThomson commented:
I suppose these will eventually show up in the L10NSharp localization dialog. I wonder whether @hatton might want us to  put a further leading element in some of the ids to group them away from the parts of the Bloom UI proper...e.g., perhaps all the template intros might want to start with "templates.", or everything that is pieces of documentation bloom pulls in might be under "doc." Since the IDs are something a user can actually see and that affects how the in-app localization is organized, I'd like his input on this one.

---

Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[DistFiles/AdobeColorProfileEULA.md, line 1 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-KrHrkGSDUl7naKIHVc9:-KrHrkGTGw63tUdfEuE-:b746z38) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/5aa593794d8ae29fb68e502c7174c6763ffead39/DistFiles/AdobeColorProfileEULA.md#L1)):*
> ```Markdown
> ### ADOBE SYSTEMS INCORPORATED {i18n="adobe.inc"}
> ```

I wonder whether this needs to be part of our own i18n system? Doesn't adobe publish the whole EULA in various languages? If so they'd probably prefer us to just use their official translations

---

*[DistFiles/license.md, line 12 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-KrHsL0U1nk-EwLoWTDw:-KrHsL0U1nk-EwLoWTDx:bul5tn0) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/5aa593794d8ae29fb68e502c7174c6763ffead39/DistFiles/license.md#L12)):*
> ```Markdown
> 
> 
> ### The MIT License {i18n="license.heading"}
> ```

Again, the MIT license is probably available in its entirety in multiple languages

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-:-KrHuGzUDH-1f82Io1Va:b2v8vii)*
<!-- Sent from Reviewable.io -->

hatton commented:
I lean towards saying we only distribute licenses in English, and never localize any of them.

hatton commented:
On the id system, I wondered about having the conversion process prepend the name of the doc... but at a cost of, if that changes, we have a hassle. So maybe we should just type a prefix that is the same throughout the document.

StephenMcConnel commented:
Review status: 7 of 14 files reviewed at latest revision, 2 unresolved discussions.

---

*[DistFiles/AdobeColorProfileEULA.md, line 1 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-KrHrkGSDUl7naKIHVc9:-KrXKrPCwHcZGPdUkAbh:b-4cz8yd) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/5aa593794d8ae29fb68e502c7174c6763ffead39/DistFiles/AdobeColorProfileEULA.md#L1)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

I wonder whether this needs to be part of our own i18n system? Doesn't adobe publish the whole EULA in various languages? If so they'd probably prefer us to just use their official translations
</blockquote></details>

I removed this file from the PR, reverting it to not be tagged for localization.

---

*[DistFiles/license.md, line 12 at r1](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-KrHsL0U1nk-EwLoWTDw:-KrXKyc1WZaREj8n2A3F:b-udxk0z) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/5aa593794d8ae29fb68e502c7174c6763ffead39/DistFiles/license.md#L12)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary><blockquote>

Again, the MIT license is probably available in its entirety in multiple languages
</blockquote></details>

I've removed this file from the PR, reverting it to not be tagged for localization.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 8 of 9 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1793#-:-Kra9IjtYytBZCP_MFcJ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

StephenMcConnel commented:
In response to John Hatton's comments, 1) I removed the two license files (reverting them to plain markdown without i18n tagging). 2) the i18n attributes used in the markdown files all used a distinct prefix value related to the original file or location.
IntegrityFailureAdvice-en.md - integrity.xxx
TrainingVideos-en.md - training.videos.xxx
Big Book/ReadMe-en.md - bigbook.xxx
Decodable Reader/ReadMe-en.md - decodable.xxx
etc.
My choices may not have been perfect, but I think they're reasonable.
